godot-core/src/builtin/string/gstring.rs

```diff
@@ -123,6 +126,22 @@ impl GString {
         std::slice::from_raw_parts(ptr as *const char, len as usize)
     }
 
+    /// Gets the internal chars slice from a [`GString`].
+    #[cfg(since_api = "4.1")]
+    pub fn chars(&self) -> &[char] {
+        unsafe {
```

Comment:
This needs a `// SAFETY: <explanation>` style comment that explains how the safety invariants are upheld. Search for this in existing code to see examples 🙂

Reply:
Since `chars_checked` doesn't have an explanation and  `chars` isn't declared `unsafe` I didn't deem it necessary.

Did you have something in mind like:
```
 /// # Safety
 ///
 /// - Godot ensures that the string is a valid UTF-32.
 ```
 or am I missunderstanding what the comment should explain?

Reply:
No, it's about the `unsafe {}` block, not any function being `unsafe`. Such blocks in Rust mean "I am aware that the thing I'm doing is not covered by the type system, and assert that it is indeed safe". It is common convention to express this in the code by writing a `// SAFETY: ` comment above the corresponding `unsafe {}` block, explaining _how_ the invariants are upheld.

This is not relevant for the caller, as it is indeed a safe function to the outside. So it should not appear in method docs.

Reply:
I hope my comment is sufficient and correct

Reply:
Should be enough, yes 🙂 this likely will need some refactoring once we remove the deprecated methods or drop support for 4.1...